lib/routes/hfut/xc/utils.ts

```diff
@@ -0,0 +1,81 @@
+import cache from '@/utils/cache';
+import got from '@/utils/got';
+import { load } from 'cheerio';
+import { parseDate } from '@/utils/parse-date';
+
+const typeMap = {
+    tzgg: { name: 'tzgg', url: 'https://xc.hfut.edu.cn/1955/list.htm', root: 'https://xc.hfut.edu.cn', title: '合肥工业大学宣城校区 - 通知公告' },
+    gztz: { name: 'gztz', url: 'https://xc.hfut.edu.cn/gztz/list.htm', root: 'https://xc.hfut.edu.cn', title: '合肥工业大学宣城校区 - 院系动态 - 工作通知' },
+};
+
+const commLink = 'https://xc.hfut.edu.cn/';
+
+const parseList = async (ctx, type) => {
+    const link = typeMap[type].url;
+    const title = typeMap[type].title;
+
+    const response = await got(link);
+    const $ = load(response.data);
+
+    const resultList = await parseArticle($);
+
+    return {
+        title,
+        link,
+        resultList,
+    };
+};
+
+async function parseArticle($) {
+    const data = $('#wp_news_w6').find('li').toArray();
+
+    const items = data.map((item) => {
+        item = $(item);
+        const oriLink = item.find('a').attr('href');
+        let linkRes = oriLink;
+        if (!oriLink.includes('http')) {
```

Comment:
The condition will also be `true` if `http` appears in the middle of `oriLink`
```suggestion
        if (!oriLink.startsWith('http')) {
```